### PR TITLE
Remove Asgardeo references from repo

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Purpose
Remove leftover Asgardeo branding and package references from the repo, completing the migration to the ThunderID identity.

### Approach

**Package metadata**
- Update root `package.json` `homepage`, `bugs`, and `repository` URLs from `asgardeo/thunder` → `thunder-id/thunderid`
- Remove `"asgardeo"` keyword from root and `frontend/package.json`

**`agent-id-sample` — frontend**
- Replace `@asgardeo/react ^0.23.6` with `@thunderid/react 0.0.5` in `package.json`
- Rename app from `asgardeo-b2c-sample-app` → `thunderid-agent-id-sample`
- Replace `AsgardeoProvider` + `platform="AsgardeoV2"` with `ThunderIDProvider`
- Replace `useAsgardeo` with `useThunderID` across all page components
- Drop hard-coded `accounts.asgardeo.io` sign-up URL construction; use `signUpUrl` from the `useThunderID` hook instead
- Update user-visible strings and comments

**`agent-id-sample` — ai-agent**
- Replace `@asgardeo/javascript ^0.9.0` with `@thunderid/javascript 0.0.5` in `package.json`
- Update comments and log strings

**`AcceptInviteBox` — test coverage**
- Make `getServerUrl` overridable per test via `vi.fn()`
- Capture `baseUrl` and `onFlowChange` props in the `AcceptInvite` mock
- Add tests for the `baseUrl` fallback to `VITE_THUNDER_BASE_URL` when `getServerUrl` returns `null`
- Add tests for `onFlowChange` setting and clearing `flowError`, and `flowError` taking priority over `error.message`

**Not changed (intentional)**
- Postman collection for Asgardeo as a federated IDP — valid supported use case
- Go IDP test fixtures using `api.asgardeo.io` URLs — generic OAuth test data
- SDK backward-compat utilities for Asgardeo URL patterns

### Related Issues
- N/A

### Related PRs
- #2783 — frontend migration (merged)
- #2805 — react-sdk-sample migration (merged)

### Checklist
- [x] Followed the contribution guidelines.
- [x] Manual test round performed and verified.
- [ ] Documentation provided.
- [x] Tests provided.
- [ ] Breaking changes.

### Security checks
- [x] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/).
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

### CODE REVIEW

File: samples/apps/agent-id-sample/frontend/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "preview": "vite preview --host 0.0.0.0"
   },
   "dependencies": {
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply samples/apps/agent-id-sample/frontend/package.json:11]: asgardeo/react is about to remove.

File: samples/apps/agent-id-sample/ai-agent/agent.ts [hunk 2]
```diff
@@ -597,14 +597,11 @@ async function getAgentTokenViaClientCredentials(): Promise<string> {
 async function createAgent() {
     console.log("##########################################################################################################");
     console.log("##      This is an Agent Authentication Flow sample application for authenticating AI agents            ##");
-    console.log("##                         using Asgardeo and LangChain framework                                       ##");
+    console.log("##                         using ThunderID and LangChain framework                                      ##");
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply samples/apps/agent-id-sample/ai-agent/agent.ts:626]: ThunderID won't be changed.

**Review by sadilchamishka — COMMENTED** [review]
Addressing review comments

File: samples/apps/agent-id-sample/ai-agent/package.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
   "license": "ISC",
   "description": "",
   "dependencies": {
-    "@asgardeo/javascript": "^0.9.0",
+    "@thunderid/javascript": "0.0.5",
     "@langchain/anthropic": "^1.3.0",
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply samples/apps/agent-id-sample/ai-agent/package.json:16]: Fixed in commit 9d5be8e0 — regenerated `package-lock.json` against `@thunderid/javascript 0.0.5`; no more `@asgardeo` references.

File: samples/apps/agent-id-sample/frontend/src/App.jsx [hunk 4]
```diff
@@ -174,9 +149,11 @@ function LiveAuthHeader() {
...
         Sign in
       </button>
-      <a className="primary-small" href={SIGN_UP_URL}>
-        Sign up
-      </a>
+      {signUpUrl && (
+        <a className="primary-small" href={signUpUrl}>
+          Sign up
+        </a>
+      )}
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply samples/apps/agent-id-sample/frontend/src/App.jsx:156]: No change needed. `signUpUrl` is populated automatically by the `@thunderid/react` SDK when OIDC discovery is enabled — `ThunderIDProvider` is already configured with `discovery: { wellKnown: { enabled: true } }` in `withConfig.tsx`, so the SDK reads the `registration_endpoint` (or equivalent) from the discovery document and surfaces it via `useThunderID().signUpUrl`. No explicit prop is required.

File: samples/apps/agent-id-sample/frontend/package.json [hunk 5]
```diff
@@ -9,7 +9,7 @@
...
-    "@asgardeo/react": "^0.23.6",
+    "@thunderid/react": "0.0.5",
     "@vitejs/plugin-react": "^4.4.1",
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply samples/apps/agent-id-sample/frontend/package.json:13]: Fixed in commit dc7da575 — all five page files (AgentPortalPage, BookingDetailsPageWithAuth, BookingsPageWithAuth, PaymentPageWithAuth, ResultsPage) were updated to import `useThunderID` from `@thunderid/react`.

File: samples/apps/agent-id-sample/ai-agent/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply samples/apps/agent-id-sample/ai-agent/package.json:16]: Fixed in commit 9d5be8e0 — same fix as the earlier thread: `package-lock.json` was regenerated against `@thunderid/javascript 0.0.5`.

File: samples/apps/agent-id-sample/ai-agent/agent.ts [hunk 6]
```diff
@@ -597,14 +597,11 @@ async function getAgentTokenViaClientCredentials(): Promise<string> {
...
     console.log("##########################################################################################################");
 
     // Call Thunder's /oauth2/token directly with the client_credentials grant.
-    // We don't use AsgardeoJavaScriptClient.getAgentToken here because the V1
-    // (legacy) Asgardeo flow expects an /oauth2/flow/embedded endpoint that
-    // Thunder doesn't expose; calling /oauth2/token directly is the same wire
-    // protocol and works for both platforms.
+    // Call Thunder's /oauth2/token directly with the client_credentials grant.
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply samples/apps/agent-id-sample/ai-agent/agent.ts:629]: Fixed in commit 9d5be8e0 — removed the duplicate comment line.

File: package.json [hunk 7]
```diff
@@ -6,16 +6,15 @@
...
   "repository": {
     "type": "git",
-    "url": "https://github.com/asgardeo/thunder"
+    "url": "https://github.com/thunder-id/thunderid"
   },
   "keywords": [
-    "asgardeo",
     "thunderid",
     "workspace",
     "wso2"
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply package.json:20]: Fixed in commit 9d5be8e0 — updated the `.audit-ignore.json` entry reason from `@asgardeo/react` to `@thunderid/react` (the advisory is for a transitive dep that still exists via the ThunderID SDK). The `npm package-lock.json` files for sample apps are regenerated in the same commit.

**Review by brionmario — DISMISSED** [review]

**Review by brionmario — APPROVED** [review]
